From fe1eb731add4a26e9e59c9c4b0ed59de91b6739c Mon Sep 17 00:00:00 2001
Message-Id: <fe1eb731add4a26e9e59c9c4b0ed59de91b6739c.1413837857.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: Deepak S <deepak.s@intel.com>
Date: Wed, 30 Jul 2014 14:48:00 +0530
Subject: [PATCH 237/312] FOR_UPSTREAM [VPG]: drm/i915: Only check modeset
 state if DPMS is enabled

No need to do a modeset_check_state while going to suspend. Also, during the
connector_state check, We raise a WARN if connector->dpms == DPMS_OFF.

WARN(connector->base.dpms == DRM_MODE_DPMS_OFF, "wrong connector dpms state\n");

Change-Id: I5f6e65b8832ea73bfea8c0baf65c400b77565046
Signed-off-by: Deepak S <deepak.s@intel.com>

Conflicts:
	drivers/gpu/drm/i915/intel_display.c
---
 drivers/gpu/drm/i915/intel_display.c |    5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index c5929d0..9c72c58 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -5610,8 +5610,9 @@ void intel_connector_dpms(struct drm_connector *connector, int mode)
 	if (connector->encoder)
 		intel_encoder_dpms(to_intel_encoder(connector->encoder), mode);
 
-	intel_modeset_check_state(connector->dev);
-	
+	if (mode == DRM_MODE_DPMS_ON)
+		intel_modeset_check_state(connector->dev);
+
 	intel_runtime_pm_put(dev_priv);
 }
 
-- 
1.7.9.5

